src/js/components/DataTable/DataTable.js

```diff
@@ -273,6 +273,40 @@ const DataTable = ({
       ? { style: { minWidth: '100%' } }
       : undefined;
 
+  const pinnedColumns = columns
+    .map(
+      (pinnedColumn, index) =>
+        pinnedColumn.pin && { property: pinnedColumn.property, id: index },
+    )
+    .filter(n => n);
+
+  const getPinnedOffset = () => {
+    const pinnedPropertiesOrder = [];
+    const pinnedColumnsProperties = {};
+    const firstRowCells = headerRef?.current?.firstChild?.cells || [];
+
+    if (firstRowCells.length !== 0) {
+      pinnedColumns.forEach(({ property, id }, index) => {
+        if (firstRowCells[id]) {
+          pinnedPropertiesOrder.push(property);
```

Comment:
If pinnedColumns becomes an array of strings, I think it would match `pinnedPropertiesOrder` and we wouldn't need this additional array.